### PR TITLE
fix: wallet analytics API request changes

### PR DESCRIPTION
**Pull request by Abhikumar98**
There are changes from backend in the request config for the analytics API, 
https://github.com/lifinance/lifi-backend/pull/2770

[API Reference](https://apidocs.li.fi/reference/get_analytics-wallets-wallet-address-1)

### CODE REVIEW

File: src/types/internal.types.ts [hunk 1]
```diff
@@ -191,3 +191,10 @@ export interface Signature {
   r: string
   s: string
 }
+
+export interface WalletAnalyticsRequest {
+  wallet_address: string
+  fromTimestamp: number
+  toTimestamp: number
+  integrator?: string
+}
```
Comment (chybisov) [inline src/types/internal.types.ts:200]: Why do we end up with a type here again? I remember we moved it to the types package.
 Also, why do we start using snake_case?

**Review by chybisov — CHANGES_REQUESTED** [review]

File: src/types/internal.types.ts (hunk 1 above)
Comment (Abhikumar98) [reply src/types/internal.types.ts:200]: This is just the request type for the endpoint, but just checked that all the request types as in /types package.

moving it there.

File: src/services/ApiService.ts [hunk 2]
```diff
@@ -522,14 +523,38 @@ const getAvailableConnections = async (
...
   )
 
+  url.searchParams.append(
+    'toTimestamp',
+    walletAnalyticsRequest.toTimestamp.toString()
+  )
+
+  if (config.integrator) {
+    url.searchParams.append('integrator', config.integrator)
+  }
```
Comment (chybisov) [inline src/services/ApiService.ts:554]: We add this automatically inside request function 🙂

**Review by chybisov — CHANGES_REQUESTED** [review]

File: src/services/ApiService.ts (hunk 2 above)
Comment (Abhikumar98) [reply src/services/ApiService.ts:554]: Inside the request method, we are passing it in headers and here the backend as a field in query params.

Seeing that, I additionally added the integrator as a query param here. If it's the same and backend always picks up from the headers then I'll go ahead and remove this.

Comment (Abhikumar98) [reply src/services/ApiService.ts:554]: Upon discussion, resorting to passing integrator via headers in request function and migrating to query param in v3 probably

**Review by chybisov — APPROVED** [review]

**Review by 0xsebn — APPROVED** [review]

**Pull request merged**